### PR TITLE
Task #230538 - [FE] - PWA - Login with keyclock

### PR DESCRIPTION
**Pull request by sonaliTekdi**
## Release Notes

- **New Features**
	- Integrated Keycloak for authentication, enhancing user login and authorization processes.
	- Added user profile handling using Keycloak tokens in the Profile screen.
	- Updated SignIn and Splash screens to utilize Keycloak for login and token management.
	- Introduced a new authentication layer in the application structure.

- **Bug Fixes**
	- Improved error handling related to token availability and decoding.

### CODE REVIEW

File: src/keycloak.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+import Keycloak from "keycloak-js";
+
+// Initialize Keycloak instance
+const keycloak = new Keycloak({
+  url: "https://dev-uba-registry-keycloak.tekdinext.com/auth", // Keycloak base URL
```
Comment (manojLondhe) [inline src/keycloak.js:6]: get these from env pls?

Comment (manojLondhe) [reply src/keycloak.js:6]: url, clientId, realm

File: src/screens/auth/Splash.tsx [hunk 2]
```diff
@@ -69,7 +80,12 @@ const Login: React.FC = () => {
           label={t("LOGIN_REGISTER_BUTTON")}
           mt={8}
         />
-        <Button className="outline-custom-btn" variant="outline" mt={2}>
+        <Button
+          className="outline-custom-btn"
+          variant="outline"
+          mt={2}
+          onClick={() => keycloak.login()}
+        >
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/screens/auth/Splash.tsx:88]: try this

File: src/screens/auth/Splash.tsx [hunk 3]
```diff
@@ -14,9 +14,20 @@ import Layout from "../../components/common/layout/Layout";
...
   const navigate = useNavigate();
   const { t } = useTranslation();
+  const { keycloak } = useKeycloak();
+  console.log(keycloak, "keycloak");
+
+  const decodedToken = keycloak?.token ? jwtDecode(keycloak.token) : null;
+  console.log("--------------------decodedToken", decodedToken);
+  if (decodedToken) {
+    navigate("/userprofile");
+  }
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/screens/auth/Splash.tsx:30]: try this

File: src/screens/auth/Splash.tsx [hunk 4]
```diff
@@ -14,9 +14,20 @@ import Layout from "../../components/common/layout/Layout";
...
 import FloatingSelect from "../../components/common/input/FloatingSelect";
 import CommonButton from "../../components/common/button/Button";
+import { useKeycloak } from "@react-keycloak/web";
+import { jwtDecode } from "jwt-decode";
+
 const Login: React.FC = () => {
   const navigate = useNavigate();
   const { t } = useTranslation();
+  const { keycloak } = useKeycloak();
+  console.log(keycloak, "keycloak");
```
[the remark this answers is not in the document]
Comment (manojLondhe) [reply src/screens/auth/Splash.tsx:24]: remove logs

[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/screens/auth/Splash.tsx:24]: removed

File: src/screens/auth/Splash.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/screens/auth/Splash.tsx:30]: Done

File: src/screens/Profile.tsx [hunk 5]
```diff
@@ -8,6 +8,8 @@ import Layout from "../components/common/layout/Layout";
 import { AuthContext } from "../utils/context/checkToken";
 import { useTranslation } from "react-i18next";
 import DocumentList from "../components/DocumentList";
+import { useKeycloak } from "@react-keycloak/web";
+import { jwtDecode } from "jwt-decode";
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/screens/Profile.tsx:12]: Done

File: src/screens/auth/SignIn.tsx [hunk 6]
```diff
@@ -28,6 +29,7 @@ const SignIn: React.FC = () => {
   const [dialogVisible, setDialogVisible] = useState(false);
   const [toastMessage, setToastMessage] = useState(false);
   const [success, setSuccess] = useState<string>("");
+  const { keycloak } = useKeycloak();
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/screens/auth/SignIn.tsx:32]: not needed

File: src/screens/Profile.tsx [hunk 7]
```diff
@@ -37,16 +39,34 @@ const UserProfile: React.FC = () => {
...
+  if (keycloak?.token) {
+    try {
+      console.log("Keycloak Token:", keycloak.token);
+
+      // Save the token in localStorage
+      localStorage.setItem("authToken", keycloak.token);
+
+      // Decode the token
+      const decodedToken = jwtDecode(keycloak.token);
+      console.log("--------------------Decoded Token", decodedToken);
```
[the remark this answers is not in the document]
Comment (sonaliTekdi) [reply src/screens/Profile.tsx:55]: Done

**Pull request merged**